avro-fastserde/src/main/java/com/linkedin/avro/fastserde/ByteBufferBackedPrimitiveFloatList.java

```diff
@@ -90,11 +91,11 @@ public static Object readPrimitiveFloatArray(Object old, Decoder in) throws IOEx
       setupElements(array, (int)totalLength);
       return array;
     } else {
-      return new PrimitiveFloatList(0);
+      return new ByteBufferBackedPrimitiveFloatList(0);
```

Comment:
will the result be mutated? maybe this can be constified?

Reply:
I think we do need to assume the result can be mutated. I'm open to introducing a deserialization mode where the returned objects are read-only though, but that would be orthogonal to this change...